x/pep/keeper/msg_server_request_private_identity.go

```diff
@@ -0,0 +1,157 @@
+package keeper
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	sdkerrors "cosmossdk.io/errors"
+	commontypes "github.com/Fairblock/fairyring/x/common/types"
+	kstypes "github.com/Fairblock/fairyring/x/keyshare/types"
+	"github.com/Fairblock/fairyring/x/pep/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	clienttypes "github.com/cosmos/ibc-go/v8/modules/core/02-client/types"
+	channeltypes "github.com/cosmos/ibc-go/v8/modules/core/04-channel/types"
+	host "github.com/cosmos/ibc-go/v8/modules/core/24-host"
+)
+
+func (k msgServer) RequestPrivateIdentity(goCtx context.Context, msg *types.MsgRequestPrivateIdentity) (*types.MsgRequestPrivateIdentityResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	reqID, found := k.GetRequestId(ctx, msg.Creator, msg.ReqId)
+	if found || len(reqID.ReqId) != 0 {
+		return nil, types.ErrReqIDAlreadyExists
+	}
+
+	k.SetRequestId(ctx, types.RequestId{
+		Creator: msg.Creator,
+		ReqId:   msg.ReqId,
+	})
+
+	requestIDStr := types.GetReqIDStr(msg.Creator, msg.ReqId)
+	req := types.PrivateRequest{
+		Creator:            msg.Creator,
+		ReqId:              requestIDStr,
+		Pubkey:             "",
+		Amount:             msg.Amount,
+		EncryptedKeyshares: make([]*commontypes.EncryptedKeyshare, 0),
+	}
+
+	k.SetPrivateRequest(ctx, req)
+
+	params := k.GetParams(ctx)
+
+	if params.IsSourceChain {
+		entry := commontypes.RequestPrivateKeyshare{
+			Creator:   msg.Creator,
+			RequestId: msg.ReqId,
+		}
+
+		k.SetPrivateReqQueueEntry(ctx, entry)
+
+		return &types.MsgRequestPrivateIdentityResponse{
+			ReqId: requestIDStr,
+		}, nil
+	} else {
+		packetData := kstypes.RequestPrivateKeysharePacketData{
+			Requester: msg.Creator,
+			RequestId: msg.ReqId,
+		}
+
+		sPort := k.GetPort(ctx)
+		timeoutTimestamp := ctx.BlockTime().Add(time.Second * 20).UnixNano()
+		_, _ = k.TransmitPrivateKeysharePacket(
+			ctx,
+			packetData,
+			sPort,
+			params.KeyshareChannelId,
+			clienttypes.ZeroHeight(),
+			uint64(timeoutTimestamp),
+		)
+
+		ctx.EventManager().EmitEvent(
+			sdk.NewEvent(
+				types.EventTypeRequestKeyshare,
```

Comment:
Is this suppose to be `EventTypePrivateKeyshareRequest`

Reply:
updated